ion/timestamp.go

```diff
@@ -515,12 +518,19 @@ func (ts Timestamp) Equal(ts1 Timestamp) bool {
 		ts.numFractionalSeconds == ts1.numFractionalSeconds
 }
 
-// TruncatedNanoseconds returns nanoseconds with trailing zeros removed (ie. 123456000 gets truncated to 123456).
+// TruncatedNanoseconds returns nanoseconds with trailing values removed up to the difference of max fractional precision - time stamp's fractional precision
+// e.g. 123456000 with fractional precision: 3 will get truncated to 123.
 func (ts Timestamp) TruncatedNanoseconds() int {
 	nsecs := ts.dateTime.Nanosecond()
-	for i := uint8(0); i < (9-ts.numFractionalSeconds) && nsecs > 0 && (nsecs%10) == 0; i++ {
-		nsecs /= 10
+
+	// Don't write coefficient byte if fractional precision is less than the nanoseconds because it is optional.
+	// e.g. A timestamp: 2021-05-25 13:41:31.00001234 with fractional precision: 2 will only write exponent byte of value 2.
```

Comment:
I don't think we need to be referencing the binary encoding here (ie. coefficient/exponent bytes).
Instead just explain that we truncate the value until it's the desired length.

Reply:
Updated comment.